### ISSUE TITLE
Finally, I find one implement of totally in original paper's way!

### ISSUE THREAD

**Issue by Lukelluke** [MingjieChen/VoiceConversionGANs#1]
Thank u sooooo much of implement this vc2 totally in original peper's way in Generator and D.

Did u tried any other way of creations with stargan-vc, for example with 80 mel bins? with stronger WaveNet
vocoder?

Hope to get any reply form u!

All the best,
Luke Huang

**Comment by MingjieChen** [comment]
Yes, I am trying to do exactly as what the stargan-vc2 paper is doing.

I haven't tried 80 mels or other vocoders, but I am willing to look at this. As I know, CycleGAN-VC3 is doing with mels. 

I have tried to incorporate StarGAN-V2 and StyleGAN2 into vc task. Maybe you can check this https://github.com/MingjieChen/LowResourceVC

**Comment by Lukelluke** [comment]
> Yes, I am trying to do exactly as what the stargan-vc2 paper is doing.
> 
> I haven't tried 80 mels or other vocoders, but I am willing to look at this. As I know, CycleGAN-VC3 is doing with mels.
> 
> I have tried to incorporate StarGAN-V2 and StyleGAN2 into vc task. Maybe you can check this https://github.com/MingjieChen/LowResourceVC

hi， thank u for ur nice job,

and i find some special difference during implement when i looked into the implement details:

1. in the Generator, the PS(pix shuffle) seems not added into the model, despite the correspond `class` function built. Does there any problem with in PS ? I noticed that u take `nn.ConvTranspose2d` to replace it. Any experiences u can all teach me~

2. In Discriminator, i'm puzzled why not use https://github.com/MingjieChen/VoiceConversionGANs/blob/1f26dec405978ad29ae0c3e974737ae6d2638d06/stgan2_ls/model.py#L344 of `x = torch.sigmoid(x)`as the final result of D, and just take ` x = torch.mean(x, dim=-1) + torch.mean(in_prod, dim = -1)  ` as final result. And about this detail, there seems no more useful imformation from source vc2 paper . I wonder why u choose this way and implement it at this way? Any thought communication will be nice~

3. What's the most question from me, is about the implement of `CIN` model in torch in voice conversion field. I searched the original paper about cin, and looked into other implementions in imaga style transfer job, such as [this CIN imple in Torch](https://github.com/ChinmayLad/neural-style-transfer/blob/dc44cd261fd9bdf684440ef104cec65fe5be15c5/normalization.py) and [this CIN imple in TensorFlow](https://github.com/MingtaoGuo/Conditional-Instance-Norm-for-n-Style-Transfer/blob/master/ops.py#L5) . Because of the difference between torch and TF, I looked into torch version more, and find that she use the `F.instance_norm(
            input, self.running_mean, self.running_var, w, b,
            self.training or not self.track_running_stats,
            self.momentum, self.eps)` 
as the inplement entry, and calculate \mu and \delta manually, and then calculate \gamma && \beta , then put those information into `torch.nn. functional.instance_norm` .

Thought i realized that we use `nn.linear()` can also make the parameters join into the whole map and make them trainable, but i still wonder, Whether or not did u know / confirm that the way we implement(just as all other implements version do ) can fulfill the original paper's thought?

4. In Generator , the end part of `conv2d(k=(5,15),c_out=35)` seems directly jump off, and directly to `c_out = 1`; And does there any profits of using GLU that u build ur self not the nn.GLU , apart from that urs would not split channels, but dose the way of GLU with `return x * torch.sigmoid(x)`  sure to be right? I'm confused, and wanna get any answer from u~

Nice to make friends with u, and if u have any thoughts , communicate with me freely~

***

Last but not least, did there any converted samples available on this version (almost full in original paper's way) with WORLD vocoder , or can u share some experiences between these versions results? From similarity / naturality / intelligibility

All the best,
Luke Huang

**Comment by MingjieChen** [comment]
Hello,

Thanks for replying.

Here are answers for  your questions:

1. PS and TransConv: 

I implemented PS but I didn't test it in my experiment. But I think it will not make a huge difference if you use PS. 

2. About torch.sigmoid(x):

I spent about 1 Month to make GAN training balanced. 
When I use torch.sigmoid(), it will cause adv loss gets 0 in about 1000 steps, which means it is gradient vanishing. So I did not use it.

3. About CIN:

According to the torch version you mentioned, I think using nn.Linear() is the same as what they do. I use one-hot as label input, they use label-index as label input.


4. About GLU:

I tried both GLU. Because I found in another CycleGAN-VC unofficial implementation they tried both. I think we should use nn.GLU() but in this model I found x*sigmoid(x) has better results.

I probably will do some more GLU tests.

5. converted results:

In my new paper https://arxiv.org/abs/2010.11646 I used this project as baseline, you might find some results.

Thanks
